python/ray/train/lightning/_lightning_utils.py

```diff
@@ -106,7 +107,14 @@ def lightning_module_state_dict(self) -> Dict[str, Any]:
         """Gathers the full state dict to rank 0 on CPU."""
         assert self.model is not None, "Failed to get the state dict for a None model!"
 
-        if _LIGHTNING_GREATER_EQUAL_2_0 and _TORCH_FSDP_AVAILABLE:
+        # Lightning < 2.1 lacks FSDP state_dict_type support.
+        # (PR: https://github.com/Lightning-AI/pytorch-lightning/pull/17623).
+        # We need this patch logic to enable FSDP checkpointing between 2.0 and 2.1.
```

Comment:
Without this fix, the state dict of checkpoint in lightning 2.0.x will be empty. 

After offline discussion, we will not raise an error since the hardcoded gathering logic doesn't contradict with the lightning behavior. Instead, we add a notice in `RayFSDPStrategy` docstring to recommend users upgrade to beyond 2.1 if they want to use FSDP.